lib/laforge/activerecord.rb

```diff
@@ -52,46 +52,48 @@ def forge_diff(**forge_options)
+      filter_loaded_data_entries(attributes: attributes, source_ids: sources, present: true).sort_by(&:priority_with_fallback).reverse.uniq(&:attribute_name).each do |data_entry|
         forged_attrs[data_entry.attribute_name] = data_entry.value
       end
 
       return forged_attrs
     end
 
     # Record several pieces of information from the same source.
-    def record_data_entries(attributes_hash, source, **data_entry_options)
+    def record_data_entries(attributes_hash, source_name, **data_entry_options)
+      source_id = DataSource.find_by(name: source_name)&.id
       attributes_hash.each do |attribute_name, value|
-        record_data_entry(attribute_name, value, source, **data_entry_options)
+        record_data_entry(attribute_name, value, source_id, **data_entry_options)
       end
     end
 
     # Record a single piece of information from a source.
     # Optionally pass a custom priority for that attribute and source at the same time.
     # Optionally pass `replace: false` to leave the existing entry for the attribute and source instead of deleting it
-    def record_data_entry(attribute_name, value, source, priority: nil, replace: true)
-      data_entries.destroy(*filter_loaded_data_entries(attributes: attribute_name, sources: source)) if replace
-      data_entries << DataEntry.new(attribute_name: attribute, value: value, source: source, priority: priority)
+    def record_data_entry(attribute_name, value, source_id, priority: nil, replace: true)
+      data_entries.destroy(*filter_loaded_data_entries(attributes: attribute_name, source_ids: source_id)) if replace
+      data_entries << DataEntry.new(attribute_name: attribute_name, value: value, source_id: source_id, priority: priority)
     end
 
     # Set and save the priority of a source for the source of a single attribute
     def update_attribute_source_priority(attribute, source, priority)
-      filter_loaded_data_entries(attributes: attribute, sources: source).each {|data_entry| data_entry.update(priority: priority) }
+      filter_loaded_data_entries(attributes: attribute, source_ids: source).each {|data_entry| data_entry.update(priority: priority) }
     end
 
     private
 
     # Returns a list of the entries matching the filters
-    def filter_loaded_data_entries(attributes: nil, sources: nil, present: nil)
+    def filter_loaded_data_entries(attributes: nil, source_ids: nil, present: nil)
       list = data_entries.to_a
+      return list if list.blank?
 
-      unless attribute.nil?
+      unless attributes.nil?
         attributes = Array.wrap(attributes).map(&:to_s)
         list.select! {|data_entry| attributes.include?(data_entry.attribute_name) }
       end
 
-      unless sources.nil?
-        sources = Array.wrap(sources).map {|source| source.id if source.is_a?(ActiveRecord::Base) }
+      unless source_ids.nil?
+        sources = Array.wrap(source_ids)
         list.select! {|data_entry| sources.include?(data_entry.source_id) }
       end
```

Comment:
Need to return the list even if no `source_ids` are given.

Reply:
It does. I think its just not seen because of how github cut it off
<img width="1378" alt="Screenshot 2023-06-27 at 2 15 03 PM" src="https://github.com/combinaut/laforge/assets/8782239/4cbee786-34eb-4da4-a057-b804690c55b0">